src/main/java/com/google/sps/servlets/TranslateServlet.java

```diff
@@ -0,0 +1,35 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/translator")
+public class TranslateServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get user input.
+    String userInput = request.getParameter("text");
```

Comment:
I had an NullPtrError getting input this way for Sentiment. if you change  `request.getParameter("text")` to `request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));` That should fix that error.

Reply:
Also for this line to work you'd need to import `Collectors`. This line should do that `import java.util.stream.Collectors;`

Reply:
Done! Thank you